### PR TITLE
Replay: Fixed GSF logging in replay

### PR DESCRIPTION
**Pull request by tridge**
This fixes WriteV logging inside replay, allowing GSF logging to work properly
needed for #15813

### CODE REVIEW

**Comment by peterbarker** [comment]
I'd already started to move these to normal log structures this-morning:

https://github.com/ArduPilot/ardupilot/pull/15767/commits/2af70a3974fd603788880cc17626f772df05e72d

**Review by priseborough — APPROVED** [review]
Checked with replay of flight log.

File: libraries/AP_Logger/AP_Logger.cpp [hunk 1]
```diff
@@ -652,9 +652,33 @@ void AP_Logger::set_vehicle_armed(const bool armed_state)
...
+        log_write_fmts = f;
+    }
+}
+#endif
+
 
 // start functions pass straight through to backend:
 void AP_Logger::WriteBlock(const void *pBuffer, uint16_t size) {
+#if APM_BUILD_TYPE(APM_BUILD_Replay)
+    save_format_Replay(pBuffer);
```
Comment (peterbarker) [inline libraries/AP_Logger/AP_Logger.cpp:680]: isn't this going to push an entry onto the list for *every single format message* - Replay message or not?

Probably want some early-return code in the top of `save_format_Replay` - to check it if it is in our static list of messages (and potentially also already in the list of dynamic messages ('though that shouldn't happen)).

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_Logger/AP_Logger.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_Logger/AP_Logger.cpp:680]: > isn't this going to push an entry onto the list for _every single format message_ - Replay message or not?

yes. That is quite deliberate and is harmless.

**Comment by peterbarker** [comment]
Merged after talking with tridge about this.  The full-list is to help us with remapping IDs into the future.

**Pull request merged**